src/test/kotlin/lotto/model/PurchasedTicketsTest.kt

```diff
@@ -0,0 +1,38 @@
+package lotto.model
+
+import org.assertj.core.api.Assertions.assertThat;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.assertThrows
+
+class PurchasedTicketsTest {
+    private val testGenerator = TestTicketNumberGenerator(listOf(1, 2, 3, 4, 5, 6))
```

Comment:
I see you design your code to be testable 👍 

This is also called the **Strategy Pattern**.
https://refactoring.guru/design-patterns/strategy
> Strategy is a behavioral design pattern that lets you define a family of algorithms, put each of them into a separate class, and make their objects interchangeable.